### PR TITLE
tensorfow examples

### PR DESCRIPTION
**Pull request by ms4952**
Created new folder tensorflow-comparison. Which refers to the material related to multi gpu tensorflow blog. Have made changes to the code to make it executable. 
 Added single gpu example,  revised previous code and added content .
Both folders have different  recipes .

### CODE REVIEW

File: examples/tensorflow-comparison/.saturn/saturn.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "example-tensorflow-comparison",
+  "image_uri": "public.ecr.aws/saturncloud/saturn-tensorflow:2022.01.06",
+  "description": "Training single node and multi GPU models in TensorFlow and drawing comparison",
```
Comment (jnolis) [inline examples/tensorflow-comparison/.saturn/saturn.json]:
```suggestion
  "description": "Compare TensorFlow model training for single and multiple GPUs",
```

File: examples/tensorflow-comparison/.saturn/saturn.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "start_script": "wandb login --relogin $WANDB_LOGIN || true",
+  "git_repositories": [
+    {
+      "url": "https://github.com/saturncloud/examples"
+    }
+  ],
+  "jupyter_server": {
+    "disk_space": "100Gi",
+    "instance_type": "8xlarge"
```
Comment (jnolis) [inline examples/tensorflow-comparison/.saturn/saturn.json]: I don't think this is the right instance type for GPU. It should use the instance type that has GPU available for hosted

File: examples/tensorflow-comparison/.saturn/saturn.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "pip": "wandb"
+  },
+  "start_script": "wandb login --relogin $WANDB_LOGIN || true",
+  "git_repositories": [
+    {
+      "url": "https://github.com/saturncloud/examples"
+    }
+  ],
+  "jupyter_server": {
+    "disk_space": "100Gi",
```
Comment (jnolis) [inline examples/tensorflow-comparison/.saturn/saturn.json]: This is probably too much disk space

File: examples/tensorflow-comparison/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Comparing training of singlenode and Multi-GPU Tensorflow.
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
# Compare Training of Single GPU and Multi-GPU TensorFlow
```

File: examples/tensorflow-comparison/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+These are the examples which were given in [this](https://saturncloud.io/blog/tensorflow_intro/) blogpost.
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion

These examples are the code that supports the Saturn Cloud [Multi-GPU TensorFlow blog post](https://saturncloud.io/blog/tensorflow_intro/).
```

File: examples/tensorflow-comparison/README.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+## Current Scripts
+
+* Training the [single node](comparison-training-singlenode.ipynb)
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
* Train a TensorFlow model with a [single GPU](comparison-training-singlenode.ipynb).
```

File: examples/tensorflow-comparison/README.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Comparing training of singlenode and Multi-GPU Tensorflow.
+These are the examples which were given in [this](https://saturncloud.io/blog/tensorflow_intro/) blogpost.
+
+## Current Scripts
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
**Included Notebooks:**
```

File: examples/tensorflow-comparison/README.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+* Training the [single node](comparison-training-singlenode.ipynb)
+* Training [multi-gpu single machine](comparison-training-multigpu.ipynb)
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
* Train a TensorFlow model with [multiple GPUs on a single machine](comparison-training-multigpu.ipynb).
```

File: examples/tensorflow-comparison/README.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Comparing training of singlenode and Multi-GPU Tensorflow.
+These are the examples which were given in [this](https://saturncloud.io/blog/tensorflow_intro/) blogpost.
+
+## Current Scripts
+
+* Training the [single node](comparison-training-singlenode.ipynb)
+* Training [multi-gpu single machine](comparison-training-multigpu.ipynb) 
+
+## Things to Watch For
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
## Notes
```

File: examples/tensorflow-comparison/README.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+* Tensorflow GPU training runs (at least) do not release memory upon concluding. These will use all or nearly all GPU memory available.
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
* TensorFlow GPU training runs do not release memory upon concluding. These will use all or nearly all GPU memory available. You can restart the Saturn Cloud resource if need to release the memory.
```

File: examples/tensorflow-comparison/README.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+* I've used Weights and Biases throughout this code to monitor GPU performance - if running this, user should set their own Saturn env credential for wandb.
```
Comment (jnolis) [inline examples/tensorflow-comparison/README.md]:
```suggestion
* Weights & Biases is used to monitor the examples. To use it, either set the environment variable `WANDB_LOGIN` as a Saturn Cloud credential, or set it manually within the notebooks.
```

File: examples/tensorflow-comparison/README.md (hunk 7 above)
Comment (jnolis) [reply examples/tensorflow-comparison/README.md]: Can we rename the notebooks:

`train-model-single-gpu.ipynb`
`train-model-multi-gpu.ipynb`

File: examples/tensorflow-comparison/comparison-training-multigpu.ipynb [hunk 12]
```diff
@@ -0,0 +1,197 @@
+{
```
Comment (jnolis) [inline examples/tensorflow-comparison/comparison-training-multigpu.ipynb:1]: This should have words, and titles, and comments. It shouldn't just be a notebook of only code.

File: examples/tensorflow-comparison/comparison-training-singlenode.ipynb [hunk 13]
```diff
@@ -0,0 +1,177 @@
+{
```
Comment (jnolis) [inline examples/tensorflow-comparison/comparison-training-singlenode.ipynb:1]: This should have titles, words, and comments. It shouldn't just be a notebook that's empty

File: examples/tensorflow-comparison/getdata.txt [hunk 14]
```diff
@@ -0,0 +1,12 @@
+### How to get the datasets for this example
```
Comment (jnolis) [inline examples/tensorflow-comparison/getdata.txt:1]: This file doesn't make sense. It should be in the readme with code chunks

**Review by jnolis — CHANGES_REQUESTED** [review]

**Comment by ms4952** [comment]
Closing this one. This is the new pr https://github.com/saturncloud/examples/pull/316

**Pull request closed without merging**